src/ui/storyTellingServerUI.js

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2017-2019 OCAD University
+Licensed under the New BSD license. You may not use this file except in compliance with this licence.
+You may obtain a copy of the BSD License at
+https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENSE.txt
+*/
+
+/* global sjrk */
+
+"use strict";
+
+/* A classic query string parser via
+ * https://stackoverflow.com/questions/901115/how-can-i-get-query-string-values-in-javascript
+ * - "name": the name of the query string variable to retrieve
+ * - "url": an optional URL to parse. Uses actual page URL if not provided
+ */
+sjrk.storyTelling.getParameterByName = function (name, url) {
+    if (!url) { url = window.location.href; }
+    if (name) { name = name.replace(/[\[\]]/g, "\\$&"); }
+    var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
+        results = regex.exec(url);
+    if (!results) { return null; }
+    if (!results[2]) { return ""; }
+    return decodeURIComponent(results[2].replace(/\+/g, " "));
+};
+
+/* Loads a story View page and a particular story from a story ID from the query string
+ * - "theme": the theme of the story View page (pass "page" in for base theme)
+ * - "options": additional options to merge into the View page
+ */
+sjrk.storyTelling.loadStoryFromParameter = function (theme, options) {
+    var storyPromise = fluid.promise();
+
+    var storyId = sjrk.storyTelling.getParameterByName("id");
+
+    if (storyId) {
+        $.get("/stories/" + storyId, function (data) {
+            var retrievedStory = JSON.parse(data);
+
+            options = options || {};
+            options.distributeOptions = {
+                "target": "{that story}.options.model",
+                "record": retrievedStory
+            };
+
+            var storyViewComponent = sjrk.storyTelling[theme].storyView(options);
+            storyPromise.resolve(storyViewComponent);
+        }).fail(function (error) {
+            storyPromise.reject(error);
+        });;
+    } else {
+        storyPromise.reject();
```

Comment:
Please provide a proper error message. Also, can we use a common structure for all returned errors:
```
{
    isError: true,
    message: {the-actual-error-message}
}
```
to distinguish errors and non-errors. This structure has already been used in [staticMiddlewareSubdirectoryFilter.js](https://github.com/BlueSlug/sjrk-story-telling/blob/SJRK-237/src/server/middleware/staticMiddlewareSubdirectoryFilter.js#L40).

Reply:
This comment applies to the entire code base. Thanks.